snippet.js

```diff
@@ -0,0 +1,16 @@
+//function call for update_color(), starting with arbitrary hex color #52beb5
+update_color("#52beb5", 10000, 10000);
+
+//function definition for update_color();
+function update_color(color, cycle_time, wait_time) {
+    setInterval(function first_color() {
+        document.body.style.backgroundColor = color;
+        setTimeout(change_color, wait_time);
+    }, cycle_time);
+
+
+    //nested function definition for change_color()
+    function change_color() {
+        document.body.style.backgroundColor = '#'+Math.floor(Math.random()*16777215).toString(16);
```

Comment:
wouldnt it be better to only set the background color in one place instead of repeating the code `document.body.style.backgroundColor =`?